OMCompiler/Compiler/Util/List.mo

```diff
@@ -284,6 +284,31 @@ algorithm
   end match;
 end isEqualOnTrue;
 
+public function allEqual<T>
+  "Takes a list and an equality function, and returns whether all elements in the list are equal."
+  input list<T> inList;
+  input CompFunc inCompFunc;
+  output Boolean outAllEqual = true;
+
+  partial function CompFunc
+    "Returns true if e1 = e2, otherwise false."
+    input T e1;
+    input T e2;
+    output Boolean res;
+  end CompFunc;
+
+protected
+  T e1 = first(inList);
```

Comment:
you should check if the list is empty before you do this. i would check `listEmpty()` in the algorithm section instead of `listLength() > 1` if the list is empty just do nothing and if the list is not empty, do what you algorithm does right now. no need to check if its `>1`, you will just iterate over nothing in that case.

Reply:
You can also match the list like this
```
e1 :: rest := inList;
```
and loop over `rest` to avoid comparing the first element against itself.

Reply:
Thank you for your suggestions for improvement. The function has been adjusted, please review it again.